gazelle {fix,update} -h panics

Running `gazelle fix -h` or `gazelle update -h` panics.
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x76478d]

goroutine 1 [running]:
github.com/bazelbuild/bazel-gazelle/internal/language/go.(*externalFlag).String(0xc4200da0f0, 0x80a900, 0xc4200da0f0)
	github.com/bazelbuild/bazel-gazelle/internal/language/go/config.go:150 +0xd
flag.isZeroValue(0xc4201ea280, 0x87245c, 0x8, 0x88cb61)
	flag/flag.go:395 +0x102
flag.(*FlagSet).PrintDefaults.func1(0xc4201ea280)
	flag/flag.go:469 +0x1b9
flag.(*FlagSet).VisitAll(0xc4200b6a80, 0xc4201ad390)
	flag/flag.go:325 +0x67
flag.(*FlagSet).PrintDefaults(0xc4200b6a80)
	flag/flag.go:478 +0x4f
main.fixUpdateUsage(0xc4200b6a80)
	github.com/bazelbuild/bazel-gazelle/cmd/gazelle/fix-update.go:334 +0xbc
main.newFixUpdateConfiguration(0x0, 0xc4200be1d0, 0x1, 0x1, 0xc4200cdf80, 0x4, 0x4, 0xc420100600, 0x3, 0x3, ...)
	github.com/bazelbuild/bazel-gazelle/cmd/gazelle/fix-update.go:264 +0xac3
main.runFixUpdate(0x0, 0xc4200be1d0, 0x1, 0x1, 0xc4200ced98, 0xc4201adf01)
	github.com/bazelbuild/bazel-gazelle/cmd/gazelle/fix-update.go:148 +0x8f9
main.run(0xc4200be1c0, 0x2, 0x2, 0xc420194690, 0xc420194660)
	github.com/bazelbuild/bazel-gazelle/cmd/gazelle/gazelle.go:77 +0x170
main.main()
	github.com/bazelbuild/bazel-gazelle/cmd/gazelle/gazelle.go:58 +0x93
```

--- a/cmd/gazelle/integration_test.go
+++ b/cmd/gazelle/integration_test.go
@@ -114,6 +114,23 @@ func runGazelle(wd string, args []string) error {
 	return run(args)
 }
 
+// TestHelp checks that help commands do not panic due to nil flag values.
+// Verifies #256.
+func TestHelp(t *testing.T) {
+	for _, args := range [][]string{
+		{"help"},
+		{"fix", "-h"},
+		{"update", "-h"},
+		{"update-repos", "-h"},
+	} {
+		t.Run(args[0], func(t *testing.T) {
+			if err := runGazelle(".", args); err != nil {
+				t.Error(err)
+			}
+		})
+	}
+}
+
 func TestNoRepoRootOrWorkspace(t *testing.T) {
 	dir, err := createFiles(nil)
 	if err != nil {

--- a/internal/flag/flag.go
+++ b/internal/flag/flag.go
@@ -32,7 +32,7 @@ func (m *MultiFlag) Set(v string) error {
 }
 
 func (m *MultiFlag) String() string {
-	if m == nil {
+	if m == nil || m.Values == nil {
 		return ""
 	}
 	return strings.Join(*m.Values, ",")
@@ -53,7 +53,7 @@ func (f *ExplicitFlag) Set(value string) error {
 }
 
 func (f *ExplicitFlag) String() string {
-	if f == nil {
+	if f == nil || f.Value == nil {
 		return ""
 	}
 	return *f.Value

--- a/internal/language/go/config.go
+++ b/internal/language/go/config.go
@@ -144,7 +144,7 @@ func (f *externalFlag) Set(value string) error {
 }
 
 func (f *externalFlag) String() string {
-	if f == nil {
+	if f == nil || f.depMode == nil {
 		return "external"
 	}
 	return f.depMode.String()

--- a/internal/language/proto/config.go
+++ b/internal/language/proto/config.go
@@ -153,7 +153,7 @@ func (f *modeFlag) Set(value string) error {
 
 func (f *modeFlag) String() string {
 	var mode Mode
-	if f != nil {
+	if f != nil && f.mode != nil {
 		mode = *f.mode
 	}
 	return mode.String()

